File: clojure/difference-of-squares/src/difference_of_squares.clj
(ns difference-of-squares)

(defn sum-of-squares [number] 
  (->>
    (range (inc number))
    (map #(* % %))
    (reduce +)))

(defn square-of-sum [number] 
  (->>
    (range (inc number))
    (reduce +)
    (#(* % %))))

(defn difference [number] 
  (- (square-of-sum number) (sum-of-squares number)))

File: kotlin/difference-of-squares/src/main/kotlin/DifferenceOfSquares.kt
class Squares(val num: Int) {

    fun sumOfSquares(): Int {
        return (1..num).sumOf { it * it }
    }

    fun squareOfSum(): Int {
        val sumOfNums = (1..num).sum()
        return sumOfNums * sumOfNums
    }

    fun difference(): Int {
        return squareOfSum() - sumOfSquares()
    }
}
